SAT-37817 - Hide Recommendations page for non-registered hosts

chris1984 opened the pull request:
#### What are the changes introduced in this pull request?
* Remove the dependency on host-specific response data
* This approach is consistent with how other components in the codebase handle this configuration and ensures that all hosts (with or without insights_facet) properly determine which advisor engine to display based on the global system settings.
* Now the Insights/Recommendations tab will be hidden if either:
  * The host is not a RHEL host (existing logic), OR
  * The host doesn't have an insights_facet (new logic)

* This means that for a host to show the Recommendations tab, it must be:
  * A RHEL host (Red Hat Enterprise Linux), AND
  * Have an insights_facet (indicated by the presence of insights_attributes in the response)

#### Considerations taken when implementing this change?
* Made sure it still works on non-iop systems

#### What are the testing steps for this pull request?
* Setup a devel box with IOP
* Click on the devel box hostname since that should not have a subscription/insights_facet and see that it pulls up the non-iop recommendations page
<img width="1920" height="1033" alt="Screenshot 2025-09-19 at 14-53-42 ip-10-0-168-128 rhos-01 prod psi rdu2 redhat com" src="https://github.com/user-attachments/assets/9b11df92-b89a-48f3-ab67-6f4c95c0a789" />

* Check out PR
* Clear your browser cache
* Visit the same host and see we have don't have a recommendations page

chris1984 committed:
SAT-37817 - Show correct Recommendations for iop non registered hosts

jeremylenz commented on webpack/InsightsHostDetailsTab/NewHostDetailsTab.js:
```diff
@@ -131,28 +131,17 @@ const IopInsightsTabWrapped = props => (
+  const isLocalAdvisorEngine = useAdvisorEngineConfig();
 
-  return isLocalAdvisorEngine ? (
+  return isLocalAdvisorEngine === true ? (
```
I agree with AI here. Is this `===` really needed? If we're sure that `isLocalAdvisorEngine` returns a Boolean (and we should be, it's our code..) then a simple `isLocalAdvisorEngine ?` should be fine here, unless I am missing something.

jeremylenz commented on webpack/InsightsHostDetailsTab/NewHostDetailsTab.js:
```diff
@@ -131,28 +131,17 @@ const IopInsightsTabWrapped = props => (
-    },
-  }),
-};
+InsightsTab.propTypes = {};
```
You can delete these propTypes

jeremylenz commented:
> * Consider renaming `isLocalAdvisorEngine` to something like `isIopModeEnabled` to better reflect that it’s driven by your `useAdvisorEngineConfig` hook.

Not in the scope of this PR, but I've been wanting to do this. It's no longer just "advisor," so we should call it `useIopConfig` and `isIop` or some such.

chris1984 commented:
> > * Consider renaming `isLocalAdvisorEngine` to something like `isIopModeEnabled` to better reflect that it’s driven by your `useAdvisorEngineConfig` hook.
> 
> Not in the scope of this PR, but I've been wanting to do this. It's no longer just "advisor," so we should call it `useIopConfig` and `isIop` or some such.

Changed the names, let me know what you think? Also updated the changes you and AI wanted.

jeremylenz reviewed (commented):
This works, but thinking about it more ..

- I don't think an unregistered host will _ever_ have recommendations.
- For Vulnerabilities, we just don't show the tab at all.

I think we should just not show the tab at all for unregistered hosts. Thoughts?

chris1984 commented:
> This works, but thinking about it more ..
> 
> * I don't think an unregistered host will _ever_ have recommendations.
> * For Vulnerabilities, we just don't show the tab at all.
> 
> I think we should just not show the tab at all for unregistered hosts. Thoughts?

Makes sense, updated the pr to:
Now the Insights/Recommendations tab will be hidden if either:

The host is not a RHEL host (existing logic), OR
The host doesn't have an insights_facet (new logic)

This means that for a host to show the Recommendations tab, it must be:

A RHEL host (Red Hat Enterprise Linux), AND
Have an insights_facet (indicated by the presence of insights_attributes in the response)

jeremylenz reviewed (changes requested):
I checked out the PR
My host has insights_attributes in the response, but I don't see a Recommendations tab:

<img width="3840" height="2096" alt="Image" src="https://github.com/user-attachments/assets/60893700-87d0-4565-80a0-9bcd8577d8b6" />

chris1984 commented:
> I checked out the PR My host has insights_attributes in the response, but I don't see a Recommendations tab:
> 
> <img alt="Image" width="2000" height="2000" src="https://private-user-images.githubusercontent.com/22042343/494048983-60893700-87d0-4565-80a0-9bcd8577d8b6.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTg4NDg2NDYsIm5iZiI6MTc1ODg0ODM0NiwicGF0aCI6Ii8yMjA0MjM0My80OTQwNDg5ODMtNjA4OTM3MDAtODdkMC00NTY1LTgwYTAtOWJjZDg1NzdkOGI2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTI2VDAwNTkwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI4YzY5OGUzODY5OGQ0MGQwZjIyZTgzMmM1MDUzNjA2YmRlMjIxNDE3MGQ1YzkwMzEwNGUyNGRmODJhNDc5NzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.3nJLIecx72gbmg8IpHYaeQW2sQElztGvl6oWetzgT4w">

@jeremylenz updated, let me know what you think. I tested it on a host that didn't have an `insights_facet` and one that did, and it worked as intended now.

jeremylenz reviewed (approved):
Works well

thanks @chris1984 !

ACK 👍
